### ISSUE TITLE
Different behavior on `index.qmd`

### ISSUE THREAD

**Issue by dleen** [fastai/nbdev#889]
I noticed some new behavior when running `nbdev_docs` when it hits the `index.qmd` file:

```
[ 1/24] 12_cli.ipynb
[ 2/24] explanations/config.ipynb
[ 3/24] migrating.ipynb
[ 4/24] 10_clean.ipynb
[ 5/24] 404.qmd
[ 6/24] 01_config.ipynb
[ 7/24] getting_started.ipynb
[ 8/24] 04b_doclinks.ipynb
[ 9/24] 03_process.ipynb
[10/24] index.qmd

Starting python3 kernel...Done

Executing 'index.ipynb'
  Cell 1/1...Done
```

I was getting an error when this file was being processed:
```
ModuleNotFoundError: No module named 'fastcore'
```

and I realised it was because it was using a different kernel.

I ran:
```
> jupyter kernelspec list                                         
Available kernels:
  python3    /Users/dleen/Library/Jupyter/kernels/python3
```
which showed it was using some kernel installed at the system level (even though I had an active conda environment).

I got rid of this using `jupyter kernelspec uninstall python3` and installed a new one: `python -m ipykernel install --sys --name python3` which indeed was part of the environment:

```
> jupyter kernelspec list                                                                             
Available kernels:
  python3    /Users/dleen/mambaforge/share/jupyter/kernels/python3
```

And it succeeded. Not sure how the `*.ipynb` notebooks are being executed, but if `index.qmd` could use the same kernel as those this wouldn't have been an issue.

**Comment by jph00** [comment]
That's interesting. `index.qmd` is a bit of an experiment - using plain quarto qmd, instead of nbdev. It's a somewhat unusual style - just one big python section, without any markdown blocks. It worked pretty well for implementing this kind of page, so I wanted to provide an example of it in our repo.

The downside is, as you discovered, that it uses jupyter to render. nbdev doesn't use jupyter at all - we use execnb. I'm not sure if there's any way to tell quarto or jupyter to use the current environment.

Longer term, we've discussed a plan with @jjallaire for quarto to support running/render .py scripts directly. The script's stdout would be the text of the rendered page. This wouldn't require jupyter at all. Alternatively, we could create an execnb "engine" for quarto by creating a lua filter and some special directive.

Many thanks for reporting your experience with this. If it turns out that `index.qmd` causes problems we'll get rid of it and go back to using index.ipynb until .py pages or an execnb engine is available.

**Comment by jph00** [comment]
I've implemented this now (#895). It's a fairly simple POC that auto-renders .py scripts that have a docstring containing frontmatter. As a result, Jupyter is no longer used when rendering the nbdev docs.
